iothub_client/src/iothubtransport_amqp_telemetry_messenger.c

```diff
@@ -981,7 +980,49 @@ static void internal_on_event_send_complete_callback(void* context, MESSAGE_SEND
                 }
                 else
                 {
-                    messenger_send_result = TELEMETRY_MESSENGER_EVENT_SEND_COMPLETE_RESULT_ERROR_FAIL_SENDING;
+                    bool isResourceLimitExceeded = false;
+                    uint32_t item_count;
+                    int ret = amqpvalue_get_list_item_count(delivery_state, &item_count);
+                    for (uint32_t i = 0; !isResourceLimitExceeded && ret == 0 && i < item_count; i++)
+                    {
+                        AMQP_VALUE delivery_state_item = amqpvalue_get_list_item(delivery_state, i);
+                        if (delivery_state_item != NULL)
+                        {
+                            AMQP_VALUE item_properties = amqpvalue_get_inplace_described_value(delivery_state_item);
+                            if (item_properties != NULL)
+                            {
+                                uint32_t item_properties_count = 0;
+                                ret = amqpvalue_get_list_item_count(item_properties, &item_properties_count);
+                                for (uint32_t t = 0; !isResourceLimitExceeded && ret == 0 && t < item_properties_count; t++)
+                                {
+                                    AMQP_VALUE item_property = amqpvalue_get_list_item(item_properties, t);
+                                    if (item_property != NULL)
+                                    {
+                                        const char* symbol_value;
+                                        int ret_sym = amqpvalue_get_symbol(item_property, &symbol_value);
+                                        if (ret_sym == 0)
```

Comment:
Any reason `ret` wasn't used here? 
What are the error conditions for `ret_sym` being non-zero? Can it indicate memory or data corruption?